picard/coverart/processing/__init__.py

```diff
@@ -53,17 +56,21 @@ def run_image_metadata_filters(metadata):
 
 def run_image_processors(data, coverartimage):
     try:
+        start_time = time.time()
         info = imageinfo.identify(data)
-        both, tags, file = get_cover_art_processors()
-        for processor in both:
-            data = processor.run(data, info, ProcessingTarget.BOTH)
-        tags_data = data
-        file_data = data
-        for processor in tags:
-            tags_data = processor.run(tags_data, info, ProcessingTarget.TAGS)
-        for processor in file:
-            file_data = processor.run(file_data, info, ProcessingTarget.FILE)
-        coverartimage.set_tags_data(tags_data)
-        coverartimage.set_external_file_data(file_data)
+        processed_data = {ProcessingTarget.BOTH: data}
+        for processor, target in get_cover_art_processors():
+            if target == ProcessingTarget.SKIP:
+                common_data = processed_data[ProcessingTarget.BOTH]
+                processed_data = {ProcessingTarget.TAGS: common_data, ProcessingTarget.FILE: common_data}
+                continue
+            processed_data[target] = processor.run(processed_data[target], info, target)
+        coverartimage.set_tags_data(processed_data[ProcessingTarget.TAGS])
```

Comment:
Ok, it seems having a generator leads to more complex code than I expected.
Also I suspect `processed_data[ProcessingTarget.TAGS]` may not be always defined.

If you prefer, go back to lists, the initial goal of my suggestion was to have simpler code, but it doesn't seem to work.

Reply:
I went back to lists, sorry about that.  I assume I will have to change this part anyways to make each queue run in a different thread.